google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkServiceManager.java

```diff
@@ -139,7 +139,15 @@ private void doWait(
       String progressMessage,
       CloudSdkStatusHandler sdkLogging) {
     CloudSdkService cloudSdkService = CloudSdkService.getInstance();
-    boolean installInProgress = isInstallInProgress();
+    SdkStatus sdkStatus = cloudSdkService.getStatus();
+    boolean installInProgress = sdkStatus == SdkStatus.INSTALLING;
```

Comment:
what if you kept the `isInstallInProgress` method that just returned this (`sdkStatus == SdkStatus.INSTALLING`)
Then, here you check:

```
if (!isInstallInProgress()....)  {
  install();
}
```
Then later, when you need to know if its in progress again, just call `isInstallInProgress()` again. Presumably `install` should internally update the state correctly so that `isInstallInProgress()` returns the correct thing. I think that would be cleaner then updating the boolean like you do here on line 149).

wdyt

Reply:
I made a few changes so we're not using flags anymore, and it looks a bit nicer this way. As for the guard, I feel we might still keep it just to be sure install() won't be called if not supported (it could break things if another SDK implementation changes things)